tests/integration/ardrive.int.test.ts

```diff
@@ -225,29 +133,75 @@ describe('ArDrive class - integrated', () => {
 		});
 
 		describe('movePublicFolder', () => {
-			it('returns the correct ArFSResult', async () => {
+			beforeEach(() => {
 				stub(arfsDao, 'getPublicDrive').resolves(stubPublicDrive);
 				stub(arfsDao, 'getPublicFolder').resolves(stubPublicFolder);
+				stub(arfsDao, 'getPublicChildrenFolderIds').resolves([stubEntityIDAltTwo]); // Uses Alt ID Two
+			});
+
+			it('throws an error if its being moved inside any of its children folders', async () => {
+				await expectAsyncErrorThrow({
+					promiseToError: arDrive.movePublicFolder({
+						folderId: stubEntityID,
+						newParentFolderId: stubEntityIDAltTwo // Alt ID Two will be in Children folder ids
```

Comment:
I think we should create a set of stubs that are known to have a hierarchical relationship and access them in a clear way, e.g.:
```
interface StubFolderHierarchy<T extends ArFSFileOrFolderEntity> {
    root: T;
    child: T;
    grandchild: T;
}
const stubFolderHierarchy = {
   root: ...,
    child: ...,
    grandchild: ...
}
```
and use like:
```
stubFolderHierarchy.root // do whatever
```